orca-keel/src/test/kotlin/com/netflix/spinnaker/orca/keel/ImportDeliveryConfigTaskTests.kt

```diff
@@ -280,6 +283,28 @@ internal class ImportDeliveryConfigTaskTests : JUnit5Minutests {
       }
     }
 
+    context("with detailed git info in payload field") {
+      val trigger = objectMapper.readValue(javaClass.getResource("/trigger.json"), Trigger::class.java)
+      fixture {
+        Fixture(
+          trigger
+        )
+      }
+
+      context("parsing git metadata") {
+        test("parses correctly") {
+          execute(mutableMapOf("sendGitInfo" to true))
+          val submittedConfig = slot<DeliveryConfig>()
+          verify(exactly = 1) {
+            keelService.publishDeliveryConfig(capture(submittedConfig))
+          }
+          val m: Any = submittedConfig.captured.getOrDefault("metadata", emptyMap<String, Any?>())!!
+          val metadata: Map<String, Any?> = objectMapper.convertValue(m)
+          expectThat(metadata["gitMetadata"]).isNotEqualTo(null)
```

Comment:
I think this could be simplified like this (`metadata` defaults to an `emptyMap()`, right?):
```suggestion
          expectThat(submittedConfig.captured.metadata["gitMetadata"]).isNotNull()
```

Reply:
I'm doing this for type safety so I don't have to use `!!`. Even if metadata defaults to an empty map I need to handle the null key there and also converting it to a map if it's stored as a string.